pyomo/contrib/gdpopt/branch_and_bound.py

```diff
@@ -37,442 +46,483 @@
+        subprob_utils = subproblem.component(
+            self.original_util_block.name)
+
+        try:
+            with SuppressInfeasibleWarning():
+                try:
+                    fbbt(subproblem, integer_tol=config.integer_tolerance)
+                except InfeasibleConstraintException:
+                    # copy variable values, even if errored
+                    copy_var_list_values(
+                        from_list=subprob_utils.algebraic_variable_list,
+                        to_list=model_utils.algebraic_variable_list,
+                        config=config, ignore_integrality=True
+                    )
+                    return float('inf'), float('inf')
+                minlp_args = dict(config.minlp_solver_args)
+                if config.minlp_solver == 'gams':
+                    elapsed = get_main_elapsed_time(self.timing)
+                    remaining = max(config.time_limit - elapsed, 1)
+                    minlp_args['add_options'] = minlp_args.get('add_options',
+                                                               [])
+                    minlp_args['add_options'].append('option reslim=%s;' %
+                                                     remaining)
+                result = SolverFactory(config.minlp_solver).solve(subproblem,
+                                                                  **minlp_args)
+        except RuntimeError as e:
+            config.logger.warning(
+                "Solver encountered RuntimeError. Treating as infeasible. "
+                "Msg: %s\n%s" % (str(e), traceback.format_exc()))
+            copy_var_list_values(  # copy variable values, even if errored
+                from_list=subprob_utils.algebraic_variable_list,
+                to_list=model_utils.algebraic_variable_list,
+                config=config, ignore_integrality=True
+            )
+            return float('inf'), float('inf')
+
+        term_cond = result.solver.termination_condition
+        if term_cond == tc.optimal:
+            assert result.solver.status is SolverStatus.ok
+            lb = result.problem.lower_bound if not obj_sense_correction else \
+                 -result.problem.upper_bound
+            ub = result.problem.upper_bound if not obj_sense_correction else \
+                 -result.problem.lower_bound
+            copy_var_list_values(
+                from_list=subprob_utils.algebraic_variable_list,
+                to_list=model_utils.algebraic_variable_list,
+                config=config,
+            )
+            return lb, ub
+        elif term_cond == tc.locallyOptimal or term_cond == tc.feasible:
+            assert result.solver.status is SolverStatus.ok
+            lb = result.problem.lower_bound if not obj_sense_correction else \
+                 -result.problem.upper_bound
+            ub = result.problem.upper_bound if not obj_sense_correction else \
+                 -result.problem.lower_bound
+            # TODO handle LB absent
```

Comment:
Is this TODO still relevant?

Reply:
I'm honestly not sure--It doesn't seem like it can be because the lower bound isn't used in that `elif` even if the solver does return one. I'll remove it (and the useless `lb` variable).

Reply:
SOunds good!